harvester/network-controller-harvester
# Sync MTU value to NAD

**Problem:** Refer to [issue 4353](https://github.com/harvester/harvester/issues/4353)


**Solution:** 
- Sync the MTU of the first vlanconfig to the new network-attachment-definition in the nad webhook mutator.
- All the vlanconfigs under the same cluster network must have the same MTU in the vlanconfig webhook validator.



**Related Issue:** https://github.com/harvester/harvester/issues/4353

**Test plan:**

- Create a new cluster network.
- Create a new vlanconfig. It's allowed to update the MTU if there is only one vlanconfig.
- Create another vlanconfig with the MTU with the first one. It will be successful.
- Create one more vlanconfig with the different MTU with the first one. It will fail.
- Modify the MTU of one of vlanconfigs. It will fail.
- Create a new VM network under the cluster network. Check the YAML. You will find it has the same MTU config with the vlanconfigs in the filed `spec.config`.
- Delete one vlanconfig and modify the MTU of the remind vlanconfig. The MTU will not be synchronized to the VM network.

**Additional context:**
Considering the time limit, it's not a thorough solution, and we will do more things on the issue https://github.com/harvester/harvester/issues/4355.
